### DISCUSSION TITLE
Trouble getting Talos set up in Proxmox on IPv6-only Network

### DISCUSSION THREAD

**Discussion by jordangarside** [siderolabs/talos#9392] [General]
I've spent quite a bit of time now trying to get a single Talos control plane node or worker installed correctly in Proxmox on my IPv6 only network.
`v1.8.0`

I figured out I can put DNS server, Gateway, and Static IP in the network config page on the installer ISO.

So I do that:

> DNS: `fd01:10:100::1:1:1:1`
> Interface: `enx921536f1f4fd`
> Mode: `Static`
> Addresses: `fd01:10:100:0:3::20/48`
> Gateway: `fd01:10:100::1`

<img width="1326" alt="image" src="https://github.com/user-attachments/assets/729bea23-35bc-4706-b20c-56b6ed39695a">

And then it seems like it's working, and I can ping it.

I go from seeing these errors:
<img width="1271" alt="image" src="https://github.com/user-attachments/assets/34e5f305-e1ac-4315-a0a5-31ad3c5a8a49">

To seeing this (looks good I think?):
<img width="1292" alt="image" src="https://github.com/user-attachments/assets/dc7d7981-36f2-431a-aaf2-c92c7866be47">

But then I start seeing a new error 
<img width="1293" alt="image" src="https://github.com/user-attachments/assets/754d8a43-5633-4999-bd89-84c1cac5542c">
Not sure if it matters? I can connect to the VM over the network at this point.

If I try to apply the config then this happens:
<img width="1254" alt="image" src="https://github.com/user-attachments/assets/59d1a661-971b-491c-a892-f67ed2ce484d">

Gives me some kind of `"no disks matched for volume"` error.
I've checked multiple times and `/dev/sda` exists, and I've also tried using `diskSelector: model: "QEMU HARDDISK"` instead and get the same error.

```
➜ talosctl -n fd01:10:100:0:3::20 disks --insecure
DEV          MODEL           SERIAL   TYPE      UUID   WWID   MODALIAS      NAME   SIZE     BUS_PATH                                                                  SUBSYSTEM          READ_ONLY   SYSTEM_DISK
/dev/loop0   -               -        UNKNOWN   -      -      -             -      76 MB    /virtual                                                                  /sys/class/block   *           
/dev/sda     QEMU HARDDISK   -        HDD       -      -      scsi:t-0x00   -      69 GB    /pci0000:00/0000:00:05.0/0000:01:01.0/virtio2/host2/target2:0:0/2:0:0:0   /sys/class/block               
/dev/sr0     QEMU DVD-ROM    -        CD        -      -      scsi:t-0x05   -      106 MB   /pci0000:00/0000:00:01.1/ata2/host1/target1:0:0/1:0:0:0                   /sys/class/block 
```

It says it starts pulling the image and then nothing happens after that (last line in previous screenshot).

Also after the 1st apply I can no longer apply any config changes:
```
➜ talosctl apply-config --insecure --nodes fd01:10:100:0:3::20 --file worker.yaml
error applying new configuration: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [fd01:10:100:0:3::20]:50000: connect: connection refused"
```

I start getting 2 network errors after the apply as well (compared to 1 before):
<img width="1290" alt="image" src="https://github.com/user-attachments/assets/ce040f24-e0cc-4dc7-be82-ae02087d0625">

I assume the new one is because of my network config not specifying priority in the route:
```
network:
    nameservers:
        - fd01:10:100::1:1:1:1
    interfaces:
        - interface: enx921536f1f4fd
        dhcp: false
        addresses:
            - fd01:10:100:0:3::20/48
        routes:
            - network: ::/0
            gateway: fd01:10:100::1
 ```
 
 Ultimately I'd like to do managed DHCPv6, but I've been trying to get static IPs working first just to test Talos out.

<img width="1325" alt="image" src="https://github.com/user-attachments/assets/131f674c-7f8e-4f14-9870-be75ad75e902">

<img width="931" alt="image" src="https://github.com/user-attachments/assets/0e0b160f-f8af-40e4-8041-5108ae4ee414">

**Comment by smira** [comment]
The error about adding route is certainly duplicate routes coming in, as you have one set in the machine config, and another via the dashboard - you don't need two of them.

The error about pulling is due to the fact that `ghcr.io` doesn't have IPv6 connectivity. We can't fix it unfortunately, but you can host a registry mirror on IPv4/6 mixed environment and use that instead.

> You can configure Talos to use that mirror instead of `ghcr.io`.

**Comment by jordangarside** [comment]
I get the duplicate route error before I apply any machine config (it
happens after I use the dashboard).
Maybe it’s from the RAs from my router after IPv6 is enabled in the dashboard?

I have nat64 and dns64 so there is no public IPv6 traffic. I actually don't
have a public IPv6 address.
Did I get a pull error?
I didn't see any error specifically, it just hangs.

Is the volume error a red herring?

Thanks!

**Reply by smira** [reply]
Might be RAs, you know it better than me, but in the end it's not the problem if it's same route.

I don't have a way to guess why pull would fail, but IPv6-only Talos machine won't be able to connect to IPv4 address I guess. The error should be printed eventually.

**Comment by jordangarside** [comment]
It's very possible the download fails/stalls, I was having some issues
earlier with my nat64 dropping packets.

Is there any timeout on the download, or would it just hang forever?

**Comment by jordangarside** [comment]
Confirmed it was packet loss issues related to my nat64 gateway.

I mostly got confused by all the other messages going on and didn't realize that the download was still being attempted.
